plugins/net_plugin/net_plugin.cpp

```diff
@@ -1232,8 +1232,8 @@ namespace eosio {
       conn->socket->async_read_some(
         conn->pending_message_buffer.get_buffer_sequence_for_boost_async_read(),
         [this,c]( boost::system::error_code ec, std::size_t bytes_transferred ) {
+          connection_ptr conn = c.lock();
           if( !ec ) {
-            connection_ptr conn = c.lock();
             if (!conn) {
```

Comment:
This `!conn` needs to be moved out of if as well otherwise your `conn-peer_name()` below could be a problem.

Reply:
no, if conn is false, the function returns immediately and won't hit my peer_name.

Reply:
nevermind. I get what you mean.